lib/slack/real_time/stores/store.rb

```diff
@@ -281,29 +302,80 @@ def initialize(attrs)
         # @see https://api.slack.com/events/group_open
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/group_open.json
         on :group_open do |data|
-          groups[data.channel].is_open = true
+          channel = private_channels[data.channel]
+          channel.is_open = true if channel
         end
 
         # A private channel was renamed.
         # @see https://api.slack.com/events/group_rename
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/group_rename.json
         on :group_rename do |data|
-          channel = groups[data.channel.id]
-          channel.name = data.channel.name if channel
+          channel = private_channels[data.channel.id]
+          channel&.merge!(data.channel)
         end
 
         # A private channel was unarchived.
         # @see https://api.slack.com/events/group_unarchive
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/group_unarchive.json
         on :group_unarchive do |data|
-          channel = groups[data.channel]
+          channel = private_channels[data.channel]
           channel.is_archived = false if channel
         end
 
         # The client has successfully connected to the server.
         # @see https://api.slack.com/events/hello
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/hello.json
-        # on :hello do |data|
+        on :hello do |_data, client|
+          # rubocop:disable Metrics/BlockNesting
+          if caches.include?(:teams)
+            team_info = client.web_client.team_info
+            team = Models::Team.new(team_info.team)
+            teams[team.id] = teams.key?(team.id) ? teams[team.id].merge(team) : team
```

Comment:
The old `rtm.start` response included `prefs` for the team. There’s a `client.web_client.team_preferences_list`, but despite what https://api.slack.com/methods/team.preferences.list says, it doesn’t seem to work with a legacy bot token. I get a `missing_scope` error. Which is strange because that token does allow the store to receive updates to individual prefs through the `team_pref_change` event.

Reply:
🤷 generally I just comment things like these out and open another issue